examples/bri-3/src/bri/merkleTree/agents/merkleTreeStorage.agent.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable } from '@nestjs/common';
+import { InjectMapper } from '@automapper/nestjs';
+import { Mapper } from '@automapper/core';
+import { PrismaService } from '../../../../prisma/prisma.service';
+import MerkleTree from 'merkletreejs';
+import { BpiMerkleTree } from '../models/bpiMerkleTree';
+
+@Injectable()
+export class MerkleTreeStorageAgent extends PrismaService {
+  constructor(@InjectMapper() private mapper: Mapper) {
+    super();
+  }
+
+  async getMerkleTreeById(id: string): Promise<BpiMerkleTree> {
+    const merkleTreeModel = await this.bpiMerkleTree.findUnique({
+      where: { id },
+    });
+
+    if (!merkleTreeModel) {
+      return null;
+    }
+
+    const merkleTree = new BpiMerkleTree(
+      merkleTreeModel.id,
+      MerkleTree.unmarshalTree(merkleTreeModel.merkleTree),
+    );
+    return this.mapper.map(merkleTree, BpiMerkleTree, BpiMerkleTree);
+  }
+
+  async createNewMerkleTree(merkleTree: BpiMerkleTree): Promise<BpiMerkleTree> {
+    const treeIn = {
+      id: merkleTree.id,
+      merkleTree: MerkleTree.marshalTree(merkleTree.merkleTree),
+    };
+    const newMerkleTreeModel = await this.bpiMerkleTree.create({
+      data: {
+        ...treeIn,
+      },
+    });
+
+    const treeOut = {
+      id: newMerkleTreeModel.id,
+      merkleTree: MerkleTree.unmarshalTree(merkleTree.merkleTree),
+    };
+    return this.mapper.map(treeOut, BpiMerkleTree, BpiMerkleTree);
+  }
+
+  async updateMerkleTree(merkleTree: BpiMerkleTree): Promise<BpiMerkleTree> {
+    const treeIn = {
```

Comment:
question:  isn't it possible to utilize mapper by defining the mapping function?

Reply:
Yeah it should be possible. will see if there is a nice solution that doesnt require creating another type of merkle tree instance. If not maybe another naming scheme fiasco. compressesTree?

Reply:
I guess we can just ditch treeIn completely and directly populate the data field in the insert and update methods

Reply:
+1 to simplify treeIn and treeOut, it is a bit confusing to read

Reply:
tree in removed and data fields populated